### PR TITLE
Adding vector graphics; add `atan2` as well. Fixed function caching.

### PR DESCRIPTION
**Pull request by ldemailly**
using golang.org/x/image/vector

adding

image.move_to
image.line_to
image.close_path
image.draw

sample generates

grol

difficult part is/was what to do with alpha when adding

### CODE REVIEW

File: extensions/images.go [hunk 1]
```diff
@@ -164,7 +171,7 @@ func createImageFunctions() { //nolint:funlen // this is a group of related func
 		Name:       "image.new",
 		MinArgs:    3,
 		MaxArgs:    3,
-		Help:       "create a new RGBA image of the name and size, image starts entirely transparent",
+		Help:       "create a new NRGBA image of the name and size, image starts entirely transparent",
```
Comment (ccoVeille) [inline extensions/images.go:174]: no one reading this will understand NRGBA :grin:

Comment (ldemailly) [reply extensions/images.go:174]: it's a search and replace, but the distinction is important, it means non pre multiplied ie you can have values of RGB higher than A while in RGBA you can't

Comment (ccoVeille) [reply extensions/images.go:174]: exactly, my point is the fact someone calling the help should get a better explanation than just an obscure acronym :grimacing:

Comment (ldemailly) [reply extensions/images.go]: fixed!

**Comment by ldemailly** [comment]
With the line with thickness demo:

grol

**Comment by ldemailly** [comment]
examples/disc.gr:

grol

**Comment by ldemailly** [comment]
circle: thickness 40

grol

thickness 1

grol

**Comment by ldemailly** [comment]
Random bezier points (discord.go version)

image

**Pull request merged**